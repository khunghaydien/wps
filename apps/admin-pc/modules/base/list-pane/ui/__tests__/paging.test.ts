import snapshotDiff from 'snapshot-diff';

import reducer, {
  actions,
  INITIALIZE,
  initialState,
  SET_CURRENT,
} from '../paging';

describe('reducer()', () => {
  it('@@init', () => {
    // @ts-ignore
    const next = reducer(undefined, { type: '@@INIT' });
    expect(snapshotDiff({}, next)).toMatchSnapshot();
  });
  it(INITIALIZE, () => {
    const next = reducer(
      initialState,
      actions.initialize(
        {
          // @ts-ignore
          page: 1,
          limitPerPage: 10,
          limit: 100,
        },
        true
      )
    );
    expect(snapshotDiff(initialState, next)).toMatchSnapshot();
  });

  it(SET_CURRENT, () => {
    const state = {
      current: 1,
      isOverLimit: false,
    };
    // @ts-ignore
    const next = reducer(state, actions.setCurrent(5));
    expect(snapshotDiff(state, next)).toMatchSnapshot();
  });
});
