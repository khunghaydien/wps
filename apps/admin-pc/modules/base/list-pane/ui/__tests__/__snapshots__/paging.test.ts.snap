// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer() @@init 1`] = `
"Snapshot Diff:
- First value
+ Second value

- Object {}
+ Object {
+   \\"current\\": 0,
+   \\"isOverLimit\\": false,
+   \\"limit\\": 0,
+   \\"limitPerPage\\": 0,
+ }"
`;

exports[`reducer() ADMIN-PC/MODULES/BASE/LIST-PANE/UI/PAGING/INITIALIZE 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
-   \\"current\\": 0,
-   \\"isOverLimit\\": false,
-   \\"limit\\": 0,
-   \\"limitPerPage\\": 0,
+   \\"current\\": undefined,
+   \\"isOverLimit\\": true,
+   \\"limit\\": 100,
+   \\"limitPerPage\\": 10,
  }"
`;

exports[`reducer() ADMIN-PC/MODULES/BASE/LIST-PANE/UI/PAGING/SET 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
-   \\"current\\": 1,
+   \\"current\\": 5,
    \\"isOverLimit\\": false,
  }"
`;
